nixos/modules/services/web-apps/tachidesk-server.nix

```
@@ -0,0 +1,235 @@
+          The path to the data directory in which Tachidesk will download scans.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "tachidesk";
+        example = "root";
+        description = mdDoc ''
+          User account under which Tachidesk runs.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "tachidesk";
+        example = "medias";
+        description = mdDoc ''
+          Group under which Tachidesk runs.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Whether to open the firewall for the port in {option}`services.tachidesk-server.config.port`.
+        '';
+      };
+
+      enableSystemTray = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          A System Tray icon will be tried to be created.
+        '';
+      };
+
+      config = {
+        ip = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          example = "127.0.0.1";
+          description = mdDoc ''
+            The ip that Tachidesk will bind to.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8080;
+          example = 4567;
+          description = mdDoc ''
+            The port that Tachidesk will listen to.
+          '';
+        };
+
+        basicAuth = {
+          enable = mkEnableOption (mdDoc ''
+            add a basic access authentication to Tachidesk-Server.
+            Enabling this option is useful when hosting on a public network/the Internet
+          '');
+
+          username = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = mdDoc ''
+              The username value that you have to provide when authenticating.
+            '';
+          };
+
+          passwordFile = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            example = "/var/secrets/tachidesk-server-password";
+            description = mdDoc ''
+              The password file containing the value that you have to provide when authenticating.
+            '';
+          };
+        };
+
+        downloadAsCbz = mkOption {
+          type = types.bool;
+          default = false;
+          description = mdDoc ''
+            Download chapters as .cbz files.
+          '';
+        };
+
+        localSourcePath = mkOption {
+          type = types.path;
+          default = cfg.dataDir;
+          defaultText = lib.literalExpression "tachidesk-server.dataDir";
+          example = "/var/data/local_mangas";
+          description = mdDoc ''
+            Path to the local source folder.
+          '';
+        };
+      };
+
+      extraConfig = mkOption {
+        type = types.submodule {
+          freeformType = with lib.types; attrsOf str;
```

The example you provide below doesn't match up with the type. According to the values the HOCON generator seems to accept, I think the `freeformType` should be more accepting.

```suggestion
          freeformType = with lib.types; attrsOf (oneOf [
            str
            path
            int
            float
            bool
          ])
```

---

Indeed it was the first time I used free form, I wasn't sure of the syntax. It's done